hybridauth.module

```diff
@@ -12,8 +12,235 @@ function hybridauth_form_alter(&$form, FormStateInterface $form_state, $form_id)
+  $files = array();
+  // Try to get cached value if allowed: $reset == FALSE.
+  if (!$reset) {
+    $cache = \Drupal::cache()->get('hybridauth_providers_files');
+    if (!empty($cache->data)) {
+      $files = $cache->data;
+    }
+  }
+
+  // Scan Providers directory if $reset == TRUE or there was no cached value.
+  if (empty($files)) {
+    $files = \Drupal::service('file_system')
+      ->scanDirectory(
+        _hybridauth_library_path() . '/hybridauth/src/Provider',
+        '/\.php$/',
+        ['key' => 'name']
+      );
+    \Drupal::cache()->set('hybridauth_providers_files', $files);
+  }
+  return $files;
+}
+
+/**
+ * Returns the path to the HybridAuth library.
+ */
+function _hybridauth_library_path() {
+  $library_path = &drupal_static(__FUNCTION__, NULL);
+
+  if (!isset($library_path)) {
+    // Check if the variable is set.
+    $library_path = \Drupal::state()->get('hybridauth_library_path', '');
+
+    if (empty($library_path)) {
+      // Use libraries module or profile directory, fallback to standard directory.
+      if (\Drupal::moduleHandler()->moduleExists('libraries')) {
+        // @todo Need check this case.
+        $library_path = libraries_get_libraries(['libraries']);
+      }
+      else {
+        $profile = drupal_get_path('profile', \Drupal::installProfile());
+        $library_path = $profile . '/libraries/hybridauth';
+        if (!file_exists($library_path)) {
+          $library_path = '../vendor/hybridauth';
+        }
+      }
+    }
+    if (file_exists($library_path . '/hybridauth/src/Hybridauth.php')) {
+    }
+    // @todo Check this condition.
+    elseif (file_exists($library_path . '/hybridauth/src/Hybridauth.php')) {
+      $library_path .= '/hybridauth';
+    }
+    else {
+      \Drupal::logger('hybridauth')->notice('HybridAuth library is missing.', []);
+      $library_path = FALSE;
+    }
+  }
+
+  return $library_path;
+}
+
+/**
+ * Pre-render callback for the providers tableselect.
+ */
+function hybridauth_admin_providers_pre_render($element) {
+  // Add weight column.
+  $element['#header']['weight'] = t('Weight');
+  foreach (array_keys($element['#options']) as $provider_id) {
+    $key = 'hybridauth_provider_' . $provider_id . '_weight';
+    $element['#options'][$provider_id]['weight'] = array(
+      'data' => \Drupal::service('renderer')->render($element[$key]),
+    );
+    unset($element[$key]);
+  }
+
+  // Assign id to the table.
+  $table_id = 'hybridauth-providers';
+  $element['#attributes'] = ['id' => $table_id];
+//  drupal_add_tabledrag($table_id, 'order', 'sibling', 'hybridauth-providers-weight');
+//  drupal_attach_tabledrag(
+//    $table_id,
+//    [
+//      'action' => 'order',
+//      'relationship' => 'sibling',
+//      'group' => 'hybridauth-providers-weight',
+//    ]
+//  );
+  return $element;
+}
+
+//
+///**
+// * Load metadata for a single provider without loading all providers.
+// */
+//function hybridauth_get_provider($name) {
```

Comment:
why we need code which are commented? if you plan to add it as next iteration then remove it form this PR.

Reply:
Commented code deleted. Thanks.